### DISCUSSION TITLE
Using switch as trigger for tooltip

### DISCUSSION THREAD

**Discussion by aishwaryaborkar** [radix-ui/primitives#2902] [Help]
Basically, I'd like to show a tooltip when hovering over a toggle (using switch from `radix-ui/react-switch`) to explain why the toggle is disabled. I'm trying to use switch as a trigger for the tooltip from `radix-ui/react-tooltip` and I've looked into similar threads (such as [this one](https://github.com/radix-ui/primitives/discussions/1166)) where it explains to spread the props but it seemingly does not work for the switch. Is there something specific that needs to be done for this particular case?

**Comment by israelsilvabarbara** [comment]
same issue here,


if i add asChild to trigger then the switch is not visible, when i remove the aschild theres an error 

In HTML, <button> cannot be a descendant of <button>.
This will cause a hydration error.

**Reply by israelsilvabarbara** [reply]
fixed wrapping the switch in a div
